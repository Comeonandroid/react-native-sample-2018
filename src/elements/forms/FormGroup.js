import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'
import { prop } from 'styled-tools'

import { View } from '../index'

const FormGroup = ({ children, top, bottom, ...rest }) => (
  <Element top={top} bottom={bottom} {...rest}>
    {children}
  </Element>
)

FormGroup.propTypes = {
  children: PropTypes.node,
  top: PropTypes.number,
  bottom: PropTypes.number
}

const Element = styled(View) `
  margin-top: ${prop('top', '15px')};
  margin-bottom: ${prop('bottom', '0px')};
`

export default FormGroup