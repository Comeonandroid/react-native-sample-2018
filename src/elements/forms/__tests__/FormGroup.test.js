import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import faker from 'faker'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import FormGroup from '../FormGroup'

describe('<FormGroup />', () => {
  itBehavesLike('aThemedComponent', { subject: <FormGroup /> })
  itBehavesLike('aComponentWithChildren', { subject: <FormGroup /> })

  it('renders correctly top prop', () => {
    const top = faker.random.number(300)
    const tree = renderer.create(<FormGroup top={top} />).toJSON()
    expect(tree.props.style[1][0].marginTop).toEqual(top)
  })

  it('renders correctly bottom prop', () => {
    const bottom = faker.random.number(300)
    const tree = renderer.create(<FormGroup bottom={bottom} />).toJSON()
    expect(tree.props.style[1][0].marginBottom).toEqual(bottom)
  })
})
