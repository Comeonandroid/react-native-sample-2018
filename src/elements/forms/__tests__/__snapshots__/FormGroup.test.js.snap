// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGroup /> dark: renders correctly 1`] = `
<View
  bottom={undefined}
  style={
    Array [
      Object {},
      Array [
        Object {
          "marginBottom": 0,
          "marginTop": 15,
        },
        undefined,
      ],
    ]
  }
  top={undefined}
/>
`;

exports[`<FormGroup /> light: renders correctly 1`] = `
<View
  bottom={undefined}
  style={
    Array [
      Object {},
      Array [
        Object {
          "marginBottom": 0,
          "marginTop": 15,
        },
        undefined,
      ],
    ]
  }
  top={undefined}
/>
`;

exports[`<FormGroup /> renders component when passed in 1`] = `
<Styled(View)>
  <Component />
</Styled(View)>
`;

exports[`<FormGroup /> renders text when passed in 1`] = `
<Styled(View)>
  Test
</Styled(View)>
`;
